### ISSUE TITLE
Support more sophisticated multipage layouts

### ISSUE THREAD

**Issue by tingerrr** [typst/typst#2941]
### Description

Support more sophisticated multipage layouts to accommodate page-spanning content with related structure. More specifically, related bits of content that must break to new pages together.

An example document where this would greatly increase usability can be seen here: https://typst.app/project/ptVHSoCtLyDGFvrOfIBgWM.

The `demo.json` file was provided by the user, who looked for a solution to get a two-page layout which is split into a side containing the Tibetan script and the phonetics and French translations on the right. They are layouted per page like regular sequential content, but content is broken en relation to its translation and by 2 pages instead of one.

My initial suggestion to use a grid would introduce undesirable spacing between individual verses of the prayers, disrupting the structure of the document. The user looked for typst to avoid the tedious manual layouting work they are currently doing in InDesign (specifically manual page breaking).

An ideal solution to this specific problem would allow specifying related bits of content, which must be broken to the next even/odd page when either of them are broken, making them always visible when viewed inside a book. This would allow the author to easily adjust styling and manually add content between related paragraphs.

The current solution has bugs and doesn't allow for manual editing easily. Perhaps columns can be extended to allow matching items in each column, similar to how I've done it manually here.

### Use Case

For book or pamphlet layouts, i.e. layouts where more than on page is viewed at a time, content breaking and layout behavior may be very closely related to how elements are related across pages. An image may be on one page while its caption is on another deliberately.

A general solution to this issue would make the creation of multipage layouts easier. While these are mostly adjusted manually, additional help from the layouter would increase readability and development speed of typst documents.

**Comment by laurmaedje** [comment]
> My initial suggestion to use a grid would introduce undesirable spacing between individual verses of the prayers, disrupting the structure of the document.

Could you elaborate on that? What exactly was the problem?

**Comment by tingerrr** [comment]
> Could you elaborate on that? What exactly was the problem?

A grid solution could produce this for a single paragraph with translation:
```
1/3 line Tibetan Paragraph | 6 line phonetics and french translations
2/3                        | 2/6
3/3                        | 3/6
undesired free space       | 4/6
undesired free space       | 5/6
undesired free space       | 6/6
```

The next Tibetan paragraph should be placed directly below the previous (save for parbreaks and such).

This is what I manually implemented by having only one cell which is grown and measured to see if both sides fit their respective columns.

**Comment by laurmaedje** [comment]
Ah, I get it. The Tibetian and the translation shouldn't be aligned on a paragraph-by-paragraph basis, but they should still break together to the next page, so that each page has the same content on both sides.

**Comment by tingerrr** [comment]
> Ah, I get it. The Tibetian and the translation shouldn't be aligned on a paragraph-by-paragraph basis, but they should still break together to the next page, so that each page has the same content on both sides.

Correct, I have updated the example to match individual paragraphs with colors.

**Comment by Enivex** [comment]
Relevant:

<https://github.com/typst/typst/issues/1865> <https://github.com/typst/typst/issues/1306>

**Comment by rwmpelstilzchen** [comment]
Also relevant: https://forum.typst.app/t/how-to-typeset-two-texts-in-parallel-on-pairs-of-facing-pages/1314

**Comment by Enivex** [comment]
> Hello. I'm writing just to add my support for this. I am shifting over to Typst because of all of its virtues over other solutions I've used for years. One of my current projects is very similar to what has been described here: an original text (in my case, Arabic) with a parallel translation, both running on the same page in columns, and break across pages together. Before I post on the user forum for ideas on how best to achieve this given the current state of affairs, I thought to ask here whether there has been any development of this since the discussion above, or whether there is any advice or future plans. Thank you.
> 

As a general rule, there likely hasn't been any development unless it's been mentioned in the issue.

**Comment by kravlost** [comment]
@Enivex  So no updates about this in the **two** **years** since that issue was raised? Shame.

**Comment by Enivex** [comment]
> @Enivex  So no updates about this in the **two** **years** since that issue was raised? Shame. 

There's plenty of issues open that I'd consider much lower hanging fruit than this. There's only so much development time available.
